src/twisted/web/test/test_http.py

```diff
@@ -3412,10 +3412,36 @@ def test_addCookieSameSite(self):
         self._checkCookie(
             b"foo=bar; SameSite=strict", b"foo", b"bar", sameSite="strict"
         )
+        self._checkCookie(
+            b"foo=bar; Secure; SameSite=none",
+            b"foo",
+            b"bar",
+            sameSite="None",
+            secure=True,
+        )
+        self._checkCookie(
+            b"foo=bar; Secure; SameSite=none",
+            b"foo",
+            b"bar",
+            sameSite="none",
+            secure=True,
+        )
 
         self.assertRaises(
             ValueError, self._checkCookie, b"", b"foo", b"bar", sameSite="anything-else"
         )
+        self.assertRaises(
```

Comment:
Can we also have a separate test here with docstring explaining what is tested here and why we should expact a failure.